llvm/labs/lab2/safarov_nurlan/replaceMulWithBitShift.cpp

```diff
@@ -0,0 +1,84 @@
+#include "llvm/IR/Function.h"
+#include "llvm/Pass.h"
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/raw_ostream.h"
+#include "llvm/Transforms/Utils/BasicBlockUtils.h"
+#include <vector>
+
+using namespace llvm;
+
+namespace {
+
+bool runPlugin(Function &F) {
+  std::vector<Instruction *> instrDeleteList;
+  for (llvm::BasicBlock &funcBlock : F) {
+    for (llvm::Instruction &instr : funcBlock) {
+      if (llvm::BinaryOperator *binOper = dyn_cast<BinaryOperator>(&instr)) {
+        if (binOper->getOpcode() == llvm::Instruction::BinaryOps::Mul) {
+          Constant *constant = dyn_cast<Constant>(binOper->getOperand(0));
```

Comment:
```suggestion
          auto *constant = dyn_cast<Constant>(binOper->getOperand(0));
```

Reply:
Also please take into account that Constant may be not only integer, but also float, etc.

Reply:
Changed. Thanks!